plugin/src/main/java/io/fabric8/maven/plugin/BuildMojo.java

```diff
@@ -241,7 +293,191 @@ private void executeOpenShiftBuild(ServiceHub hub, ImageConfiguration imageConfi
+            if (!Files.isFile(manifest)) {
+                throw new MojoFailureException("No such generated manifest file: " + manifest);
+            }
+
+            String namespace = clusterAccess.getNamespace();
+            Controller controller = new Controller(client);
+
+            boolean updated = false;
+            Set<HasMetadata> entities = loadResources(client, controller, namespace, manifest, project, log);
+            for (HasMetadata entity : entities) {
+                if (entity instanceof ImageStream) {
+                    ImageStream is = (ImageStream) entity;
+                    String imageStreamName = KubernetesHelper.getName(is);
+                    if (Objects.equals(buildConfigName, imageStreamName)) {
+                        if (updateImageStreamTag(client, imageConfig, is, buildConfigName)) {
+                            updated = true;
+                        }
+                    }
+                }
+            }
+            if (updated) {
+                // lets store the entities again!
+                KubernetesList entity = new KubernetesListBuilder().withItems(new ArrayList<>(entities)).build();
+                File resourceFileBase = new File(this.targetDir, ResourceClassifier.OPENSHIFT.getValue());
+                AbstractResourceMojo.writeResourcesIndividualAndComposite(entity, resourceFileBase, ResourceFileType.yaml, log);
+            }
+        } catch (KubernetesClientException e) {
+            KubernetesResourceUtil.handleKubernetesClientException(e, this.log);
+        } catch (MojoExecutionException e) {
+            throw e;
+        } catch (Exception e) {
+            throw new MojoExecutionException(e.getMessage(), e);
+        }
+
+    }
+
+    private boolean updateImageStreamTag(OpenShiftClient client, ImageConfiguration imageConfig, ImageStream is, String buildConfigName) throws MojoExecutionException {
+        String namespace = client.getNamespace();
+        String imageName = imageConfig.getName();
+        String label = getImageLabel(imageName);
+        ImageStream currentImageStream = client.imageStreams().withName(buildConfigName).get();
+        if (currentImageStream == null) {
+            throw new MojoExecutionException("Could not find a current ImageStream with name " + buildConfigName + " in namespace " + namespace);
+        }
+        String tagSha = findTagSha(currentImageStream);
+        String name = buildConfigName + "@" + tagSha;
+        String kind = "ImageStreamImage";
+
+        ImageStreamSpec spec = is.getSpec();
+        if (spec != null) {
+            spec = new ImageStreamSpec();
+            is.setSpec(spec);
+        }
+        List<TagReference> tags = spec.getTags();
+        if (tags == null) {
+            tags = new ArrayList<>();
+            spec.setTags(tags);
+        }
+        TagReference tag = null;
+        if (tags.isEmpty()) {
+            tag = new TagReferenceBuilder().build();
```

Comment:
yeah  -left over from a refactor where I was adding logic into the builder ;) removed...